specs/core/0_beacon-chain.md

```diff
@@ -724,49 +737,94 @@ def min_empty_validator(validators: List[ValidatorRecord], current_slot: int):
 ```
 
 ```python
-def get_fork_version(state: State, slot: int) -> int:
-    return state.pre_fork_version if slot < state.fork_slot_number else state.post_fork_version
+def get_fork_version(pre_fork_version: int,
+                     post_fork_version: int,
+                     fork_slot_number: int,
+                     slot: int) -> int:
+    return pre_fork_version if slot < fork_slot_number else post_fork_version
     
-def get_domain(state: State, slot: int, base_domain: int) -> int:
-    return get_fork_version(state, slot) * 2**32 + base_domain
+def get_domain(pre_fork_version: int,
+               post_fork_version: int,
+               fork_slot_number: int,
+               slot: int,
+               base_domain: int) -> int:
+    return get_fork_version(
+        pre_fork_version,
+        post_fork_version,
+        fork_slot_number,
+        slot
+    ) * 2**32 + base_domain
```

Comment:
I like the idea of making most methods take smaller parts of the state if at all possible, but I think this one is excessive; it gets unwieldy with this many arguments.

Reply:
Agreed! I think it's fine to pass `BeaconState` in the `get_*` helper functions because the function name imply that it's a pure function. But in this case, it's because `on_startup()` calls `get_new_validators()` *before* `state` is created; and then `get_new_validators()` calls `get_domain()`... so I think it's easier to fix it by just passing the integer parameters.

Reply:
Aah, I see, good point. I would say resolve this by creating a new object `ForkData: { pre_fork_version: uint64, post_fork_version: uint64, fork_slot_number: uint64}`, so we only need to pass one thing around.

Reply:
Done!